libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2278,27 +2316,32 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			auto const& toStructType = dynamic_cast<StructType const &>(_to);
 			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 
-			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
-			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
-			solUnimplementedAssert(fromStructType.location() != DataLocation::Memory, "");
-
-			if (fromStructType.location() == DataLocation::CallData)
+			if (fromStructType.location() == toStructType.location() && toStructType.isPointer())
 			{
-				body = Whiskers(R"(
-					converted := <abiDecode>(value, calldatasize())
-				)")("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
-					{&toStructType}
-				)).render();
+				solUnimplementedAssert(toStructType.sizeOnStack() == 1, "");
```

Comment:
```suggestion
				solUnimplementedAssert(toStructType.sizeOnStack() == 1 && fromStructType.sizeOnStack() == 1, "");
```
?

Reply:
And actually this rather belongs in L2184 right when the function signature is created instead...